### PR TITLE
Avoid 'getBean' method in 'Infinispan*CacheManagerChecker' classes #38

### PR DESCRIPTION
**Pull request by lburgazzoli**

### CODE REVIEW

File: infinispan-spring-boot-starter-remote/src/main/java/org/infinispan/spring/starter/remote/InfinispanRemoteAutoConfiguration.java [hunk 1]
```diff
@@ -103,4 +110,51 @@ public RemoteCacheManager remoteCacheManager() throws IOException {
...
+
+   public static class ConditionalOnCacheType implements Condition {
+      @Override
+      public boolean matches(ConditionContext ctx, AnnotatedTypeMetadata atm) {
+         String cacheType = ctx.getEnvironment().getProperty("spring.cache.type");
+         return cacheType == null || CacheType.INFINISPAN.name().equalsIgnoreCase(cacheType);
+      }
+   }
+
+   public static class ConditionalOnConfiguration extends AnyNestedCondition {
```
Comment (slaskawi) [inline infinispan-spring-boot-starter-remote/src/main/java/org/infinispan/spring/starter/remote/InfinispanRemoteAutoConfiguration.java]: Could you please move this above `ConditionalOnCacheType`?

File: infinispan-spring-boot-starter-remote/src/test/java/test/org/infinispan/spring/starter/remote/CustomConfigurerTest.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    classes = {
+        CustomConfigurerTest.TestConfiguration.class,
+        InfinispanRemoteAutoConfiguration.class
+    },
+    properties = {
+        "spring.main.banner-mode=off",
+        "myServerList=localhost:6667"
+    }
+)
+public class CustomConfigurerTest {
```
Comment (slaskawi) [inline infinispan-spring-boot-starter-remote/src/test/java/test/org/infinispan/spring/starter/remote/CustomConfigurerTest.java]: Perhaps we could come up with a better name here? Something like `CustomConfigurerWithPropertyInjectionTest`?

**Comment by lburgazzoli** [comment]
I'm in PTO till next week and I do not know if I'll find time to rework the PR so feel free to merge and fix it :)

**Comment by lburgazzoli** [comment]
Done

**Comment by slaskawi** [comment]
Thanks @lburgazzoli !

**Comment by lburgazzoli** [comment]
did you merge it ?

**Comment by slaskawi** [comment]
@lburgazzoli ooook crap... Sorry I need another coffee apparently...

**Comment by slaskawi** [comment]
@lburgazzoli Fixed. Thanks.

**Pull request merged**